src/main/java/simpleRest/service/UserService.java

```diff
@@ -0,0 +1,59 @@
+package simpleRest.service;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+import simpleRest.entity.User;
+import simpleRest.exception.UserForbiddenException;
+import simpleRest.repository.RoleRepository;
+import simpleRest.repository.UserRepository;
+
+@RequiredArgsConstructor
+@Service
+public class UserService {
+
+    private final UserRepository userRepo;
+    private final RoleRepository roleRepository;
+    private final CookieService cookieService;
+    private final PasswordEncoder passwordEncoder;
+
+    public User save(User user) {
+        userRepo.saveAndFlush(user);
+        return user;
+    }
+
+    public User userAuthentication(String username, String password, String role)
```

Comment:
The best way is creating Authentication service for all authenticate things similar like Spring Security.